hyperspy/axes.py

```diff
@@ -49,7 +49,7 @@
 class ndindex_nat(np.ndindex):
 
     def __next__(self):
-        return super(ndindex_nat, self).next()[::-1]
+        return super(ndindex_nat, self).__next__()[::-1]
```

Comment:
``next(super(ndindex_nat, self))[::-1]``?

Reply:
This is the first thing I have tried and it doesn't work...

Here is a minimal example to reproduce:
```python
import hyperspy.api as hs

s = hs.signals.Signal1D(np.arange(10*10*10).reshape((10, 10, 10)))
[coordinate for coordinate in s.axes_manager._am_indices_generator()]
```
```python
Traceback (most recent call last):

  File "<ipython-input-3-181577425fbe>", line 1, in <module>
    [coordinate for coordinate in s.axes_manager._am_indices_generator()]

  File "<ipython-input-3-181577425fbe>", line 1, in <listcomp>
    [coordinate for coordinate in s.axes_manager._am_indices_generator()]

  File "/home/eric/Dev/hyperspy/hyperspy/axes.py", line 52, in __next__
    return next(super(ndindex_nat, self))[::-1]

TypeError: 'super' object is not an iterator
```

Reply:
Interesting, it seems then that ``super`` is evaluated after ``next``.